source/common/quic/active_quic_listener.cc

```diff
@@ -175,41 +176,9 @@ uint32_t ActiveQuicListener::destination(const Network::UdpRecvData& data) const
     return worker_index_;
   }
 
-  // This implementation is not as performant as it could be. It will result in most packets being
+  // Taking this path is not as performant as it could be. It means most packets are being
   // delivered by the kernel to the wrong worker, and then redirected to the correct worker.
-  //
-  // This could possibly be improved by keeping a global table of connection IDs, so that a new
-  // connection will add its connection ID to the table on the current worker, and so packets should
-  // be delivered to the correct worker by the kernel unless the client changes address.
-
-  // This is a re-implementation of the same algorithm written in BPF in
-  // ``ActiveQuicListenerFactory::createActiveUdpListener``
-  const uint64_t packet_length = data.buffer_->length();
-  if (packet_length < 9) {
-    return worker_index_;
-  }
-
-  uint8_t first_octet;
-  data.buffer_->copyOut(0, sizeof(first_octet), &first_octet);
-
-  uint32_t connection_id_snippet;
-  if (first_octet & 0x80) {
-    // IETF QUIC long header.
-    // The connection id starts from 7th byte.
-    // Minimum length of a long header packet is 14.
-    if (packet_length < 14) {
-      return worker_index_;
-    }
-
-    data.buffer_->copyOut(6, sizeof(connection_id_snippet), &connection_id_snippet);
-  } else {
-    // IETF QUIC short header, or gQUIC.
-    // The connection id starts from 2nd byte.
-    data.buffer_->copyOut(1, sizeof(connection_id_snippet), &connection_id_snippet);
-  }
-
-  connection_id_snippet = htonl(connection_id_snippet);
-  return connection_id_snippet % concurrency_;
+  return select_connection_id_worker_(*data.buffer_, concurrency_, worker_index_);
```

Comment:
Check `select_connection_id_worker_ != nullptr`?

Reply:
I mean surround it with an if condition checking select_connection_id_worker_ != nullptr.

Reply:
And then return what, if it is nullptr?

Reply:
I agree it makes more sense to assert non-null in constructor